django/db/models/query.py

```diff
@@ -1527,11 +1531,30 @@ def difference(self, *other_qs):
             return self
         return self._combinator_query("difference", *other_qs)
 
+    def select_for_share(self, nowait=False, skip_locked=False, of=(), key=False):
+        """
+        Return a new QuerySet instance that will select objects with a
+        FOR SHARE lock.
+        """
+        if self.query.select_for_update:
+            raise TypeError("Cannot call select_for_share() after select_for_update().")
+        if nowait and skip_locked:
+            raise ValueError("The nowait option cannot be used with skip_locked.")
+        obj = self._chain()
+        obj.query.select_for_share = True
+        obj.query.select_for_share_nowait = nowait
+        obj.query.select_for_share_skip_locked = skip_locked
+        obj.query.select_for_share_of = of
+        obj.query.select_for_key_share = key
+        return obj
+
     def select_for_update(self, nowait=False, skip_locked=False, of=(), no_key=False):
```

Comment:
Hmm. As we're adding `select_for_share()` it'd probably make sense for it to take keyword arguments only given that most arguments are booleans.. But that might be confusing if that were not the case for `select_for_update()` too. I guess we'd have to have a deprecation on that though if we wanted to enforce it. Your thoughts @felixxm?